liberum_gui/src/event_handler.rs

```diff
@@ -0,0 +1,211 @@
+
+        self.rt.block_on(async {
+            self.to_daemon_sender
+                .send(DaemonRequest::StartNode {
+                    node_name: name.to_string(),
+                })
+                .await?;
+
+            match self.from_daemon_receiver.recv().await {
+                Some(r) => info!(response = format!("{r:?}"), "Daemon responds: {:?}", r),
+                None => {
+                    error!("Failed to receive response");
+                }
+            }
+
+            anyhow::Ok(())
+        })?;
+
+        Ok(())
+    }
+
+    pub fn stop_node(&mut self, name: &str) -> Result<()> {
+        debug!(name = name.to_string(), "Trying to stop node");
+
+        self.rt.block_on(async {
+            self.to_daemon_sender
+                .send(DaemonRequest::StopNode {
+                    node_name: name.to_string(),
+                })
+                .await?;
+
+            match self.from_daemon_receiver.recv().await {
+                Some(r) => info!(response = format!("{r:?}"), "Daemon responds: {:?}", r),
+                None => {
+                    error!("Failed to receive response");
+                }
+            }
+
+            anyhow::Ok(())
+        })?;
+
+        Ok(())
+    }
+
+    pub fn create_node(&mut self, name: &str) -> Result<()> {
+        debug!(name = name.to_string(), "Trying to create node");
+
+        self.rt.block_on(async {
+            self.to_daemon_sender
+                .send(DaemonRequest::NewNode {
+                    node_name: name.to_string(),
+                    id_seed: None,
+                })
+                .await?;
+
+            match self.from_daemon_receiver.recv().await {
+                Some(r) => info!(response = format!("{r:?}"), "Daemon responds: {:?}", r),
+                None => {
+                    error!("Failed to receive response");
+                }
+            }
+
+            anyhow::Ok(())
+        })?;
+
+        Ok(())
+    }
+
+    pub fn publish_file(&mut self, node_name: &str, file_path: &Path) -> Result<String> {
+        debug!(
+            name = node_name.to_string(),
+            path = file_path.display().to_string(),
+            "Trying to publish file"
+        );
+
+        self.rt.block_on(async {
+            self.to_daemon_sender
+                .send(DaemonRequest::PublishFile {
+                    node_name: node_name.to_string(),
+                    path: file_path.to_path_buf(),
+                })
+                .await?;
+
+            match self.from_daemon_receiver.recv().await {
+                Some(r) => {
+                    match r {
+                        Ok(DaemonResponse::FilePublished { id }) => return Ok(id),
+                        Err(e) => {
+                            error!(err = e.to_string(), "Error ocurred while publishing file!");
+                            bail!("Error occured while publishing file: {}", e.to_string());
```

Comment:
Yeah, but I want to return an error and print it at the same time. I would leave it like that, because the first one uses structural logging so it would be hard to extract the common string part (not worth it).